pymodm/base/options.py

```diff
@@ -86,8 +86,14 @@ def get_field_from_attname(self, attname):
 
     def add_field(self, field_inst):
         """Add or replace a given Field."""
-        orig_field = (self.get_field(field_inst.mongo_name) or
-                      self.get_field_from_attname(field_inst.attname))
+        try:
+            orig_field = self.get_field(field_inst.mongo_name)
+        except Exception:
+            # FieldDoesNotExist, etc. may be raised by subclasses.
+            try:
+                orig_field = self.get_field_from_attname(field_inst.attname)
```

Comment:
Hold up, if we don't call `self.get_field_from_attname(field_inst.attname)`, we'll never remove the original field from the parent class, right? If the tests still may without that line perhaps we need to add such a test?